internal/context.go

```diff
@@ -238,10 +238,10 @@ func propagateCancel(parent Context, child canceler) {
 	} else {
 		go func() {
 			s := NewSelector(parent)
-			s.AddReceive(parent.Done(), func(c Channel, more bool) {
+			s.AddReceive(parent.Done(), func(c ReadChannel, more bool) {
```

Comment:
It seems `ReceiveChannel` and `SendChannel` are better names and they match go terminology.